js/main.js

```diff
@@ -77,19 +76,24 @@ function domContentLoadedListener(event) {
 
 var entriesNavItem = document.querySelector('#entry-nav');
 var entriesNewButton = document.querySelector('#new-button');
-var entryFormContainer = document.querySelector('[data-view="entry-form"]');
-var entriesMainContainer = document.querySelector('[data-view="entries"]');
 
-entriesNewButton.addEventListener('click', entriesNewButtonHandler);
+entriesNewButton.addEventListener('click', dataViewHandler);
+entriesNavItem.addEventListener('click', dataViewHandler);
 
-function entriesNewButtonHandler(event) {
-  entryFormContainer.className = 'container';
-  entriesMainContainer.className = 'container hidden';
-}
+var tabList = document.querySelectorAll('.tab');
 
-entriesNavItem.addEventListener('click', entriesNavItemHandler);
+function dataViewHandler(event) {
+  var dataView = event.target.getAttribute('data-view');
+  switchViews(dataView);
+}
 
-function entriesNavItemHandler(event) {
-  entryFormContainer.className = 'container hidden';
-  entriesMainContainer.className = 'container';
+function switchViews(string) {
+  for (var i = 0; i < tabList.length; i++) {
+    if (string === tabList[i].getAttribute('data-view')) {
+      tabList[i].className = 'tab container';
+      data.view = string;
+    } else {
+      tabList[i].className = 'tab container hidden';
```

Comment:
The `tab` class name had appropriate meaning in the `javascript-view-swapping` exercise since it was being applied to tabs that were clickable, but in this case, there is no `tab`, so this is inaccurate. You are using this class to identify view containers, so I suggest renaming this to `view` or `view-container`